test/erc20.spec.ts

```diff
@@ -1,128 +1,252 @@
+    // Deployers
+    walletL1: Signer,
+    walletL2: Signer,
+    // Additional wallets/signers
+    walletL1_2: Signer,
+    walletL1_3: Signer,
+    walletL2_2: Signer,
+    walletL2_3: Signer,
+    // L1 <> L2 communication
+    watcher: any
+
 
   const name = 'Some Really Cool Token Name'
-  const initialSupply = 10000000
+  const initialSupplyL1 = ethers.BigNumber.from('10000000') // For L1 contract
+  const initialSupplyL2 = ethers.BigNumber.from('10000000') // For L2 contract
 
-  let ERC20: Contract
-  beforeEach(async () => {
-    ERC20 = await (await ethers.getContractFactory('ERC20'))
-      .connect(account1)
-      .deploy(initialSupply, name)
-  })
+  before('set up providers and wallets', () => {
+    providerL1 = new ethers.providers.JsonRpcProvider(process.env.L1_WEB3_URL)
+    providerL2 = new ethers.providers.JsonRpcProvider(process.env.L2_WEB3_URL)
 
-  describe('the basics', () => {
-    it('should have a name', async () => {
-      expect(await ERC20.name()).to.equal(name)
-    })
+    walletL1 = new ethers.Wallet(process.env.USER_PRIVATE_KEY, providerL1)
+    walletL2 = new ethers.Wallet(process.env.USER_PRIVATE_KEY, providerL2)
 
-    it('should have a total supply equal to the initial supply', async () => {
-      expect(await ERC20.totalSupply()).to.equal(initialSupply)
-    })
+    const privateKey3 = ethers.Wallet.createRandom().privateKey
+    const privateKey4 = ethers.Wallet.createRandom().privateKey
+
+    walletL1_2 = new ethers.Wallet(privateKey3, providerL1)
+    walletL1_3 = new ethers.Wallet(privateKey4, providerL1)
+
+    walletL2_2 = new ethers.Wallet(privateKey3, providerL2)
+    walletL2_3 = new ethers.Wallet(privateKey4, providerL2)
 
-    it("should give the initial supply to the creator's address", async () => {
-      expect(await ERC20.balanceOf(await account1.getAddress())).to.equal(
-        initialSupply
-      )
-    })
   })
 
-  describe('transfer', () => {
-    it('should revert when the sender does not have enough balance', async () => {
-      const sender = account1
-      const recipient = account2
-      const amount = initialSupply + 2500000
-
-      await expect(
-        ERC20.connect(sender).transfer(await recipient.getAddress(), amount)
-      ).to.be.revertedWith(
-        "You don't have enough balance to make this transfer!"
-      )
+  /** Should we remove Watcher? */
+  // before('set up watchers', async () => {
+  //   const response: any = await fetch('http://localhost:8080/addresses.json')
+  //   const addresses = response.data
+
+  //   watcher = new Watcher({
+  //     l1: {
+  //       provider: providerL1,
+  //       messengerAddress: addresses['Proxy_OVM_L1CrossDomainMessenger'],
+  //     },
+  //     l2: {
+  //       provider: providerL2,
+  //       messengerAddress: '0x4200000000000000000000000000000000000007',
+  //     }
+  //   })
+  // })
+
+  // after('exit', async () => {
+  //   // TODO: Optimism watchers leave the process open, so we explicitely kill it
```

Comment:
This problem was fixed here: https://github.com/ethereum-optimism/optimism-monorepo/pull/404

Reply:
thanks for this! Although, this comment was copied directly from Synthetix's develop branch in their core CLI

Reply:
I'll make sure some notes to next time to clarify commented sections